cryptography/src/secp256r1/recoverable.rs

```diff
@@ -0,0 +1,1230 @@
+            commonware_utils::from_hex_formatted(
+                "519b423d715f8b581f4fa8ee59f4771a5b44c8130b4e3eacca54a56dda72b464",
+            )
+            .unwrap()
+            .as_ref(),
+        )
+        .unwrap();
+        let public_key: PublicKey = private_key.clone().into();
+        let message = commonware_utils::from_hex_formatted(
+            "5905238877c77421f73e43ee3da6f2d9e2ccad5fc942dcec0cbd25482935faaf416983fe165b1a045e
+            e2bcd2e6dca3bdf46c4310a7461f9a37960ca672d3feb5473e253605fb1ddfd28065b53cb5858a8ad28175bf
+            9bd386a5e471ea7a65c17cc934a9d791e91491eb3754d03799790fe2d308d16146d5c9b0d0debd97d79ce8",
+        )
+        .unwrap();
+        let signature = private_key.sign(None, &message);
+        assert_eq!(SIGNATURE_LENGTH, signature.len());
+        assert!(public_key.verify(None, &message, &signature));
+    }
+
+    #[test]
+    fn test_decode_zero_signature_fails() {
+        let result = Signature::decode(vec![0u8; SIGNATURE_LENGTH].as_ref());
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_decode_high_s_signature_fails() {
+        let (private_key, _) = vector_keypair_1();
+        let message = b"edge";
+        let signature = private_key.sign(None, message);
+        let mut bad_signature = signature.to_vec();
+        bad_signature[33] |= 0x80; // force S into upper range (first byte of S)
+        assert!(Signature::decode(bad_signature.as_ref()).is_err());
+    }
+
+    #[test]
+    fn test_decode_zero_r_signature_fails() {
+        let (private_key, _) = vector_keypair_1();
+        let message = b"edge";
+        let signature = private_key.sign(None, message);
+        let mut bad_signature = signature.to_vec();
+        for b in bad_signature.iter_mut().skip(1).take(32) {
+            *b = 0x00;
+        }
+        // ensure S component is non-zero
+        bad_signature[33] = 1;
+        assert!(Signature::decode(bad_signature.as_ref()).is_err());
+    }
+
+    // Ensure RFC6979 compliance (should also be tested by underlying library)
+    #[test]
+    fn test_rfc6979() {
+        let private_key: PrivateKey = PrivateKey::decode(
+            commonware_utils::from_hex_formatted(
+                "c9afa9d845ba75166b5c215767b1d6934e50c3db36e89b127b8a622b120f6721",
+            )
+            .unwrap()
+            .as_ref(),
+        )
+        .unwrap();
+
+        let (message, exp_sig) = (
+            b"sample",
+            p256::ecdsa::Signature::from_slice(
+                &commonware_utils::from_hex_formatted(
+                    "efd48b2aacb6a8fd1140dd9cd45e81d69d2c877b56aaf991c34d0ea84eaf3716
+                    f7cb1c942d657c41d436c7a1b6e29f65f3e900dbb9aff4064dc4ab2f843acda8",
+                )
+                .unwrap(),
+            )
+            .unwrap(),
+        );
+        let signature = private_key.sign(None, message);
+        assert_eq!(
+            signature.signature.to_bytes().to_vec(),
```

Comment:
how is this passing 🤔

Reply:
Yep; We no longer have the accessor, so we just directly access it within the test.

Reply:
https://github.com/commonwarexyz/monorepo/pull/2116#discussion_r2496290530